### DISCUSSION TITLE
Song request API - Problem with IP on  server side requests

### DISCUSSION THREAD

**Discussion by RM-FM** [AzuraCast/AzuraCast#6795] [General]
Hi,

We use Wordpress wp_remote_post() function to call the API endpoint /station/{station_id}/request/{request_id}. This is working perfectly fine but as the reques is executed service sided, it provides the local server IP but not the original client IP in the API request. As a consequence, each request is logged under the server IP. Is there a way to pass an addtional parameter or header field to the API that holds the client IP? I already tried things like "X-Forwarded-For" and "X-Client-IP".

**Comment by BusterNeece** [comment]
@RM-FM If you authenticate the request (i.e. pass a valid API key, i.e. via `Authorization: Bearer (api-key)` or `X-API-Key: api-key`) it won't rate-limit the submissions by IP.
